roles/docker/tasks/main.yml

```diff
@@ -10,11 +10,11 @@
 - name: Downgrade docker if necessary
   command: "{{ ansible_pkg_mgr }} downgrade -y docker-{{ docker_version }}"
   register: docker_downgrade_result
-  when: not docker_version_result | skipped and docker_version_result | default('0.0', True) | version_compare(docker_version, 'gt')
+  when: not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'gt')
 
 - name: Install docker
-  action: "{{ ansible_pkg_mgr }} name=docker{{ '-' + docker_version if docker_version != '' else '' }} state=present"
-  when: not openshift.common.is_atomic | bool and not docker_downgrade_result | changed
+  action: "{{ ansible_pkg_mgr }} name=docker{{ '-' + docker_version if docker_version_result.stdout == '' }} state=latest"
+  when: not openshift.common.is_atomic | bool and not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'lt')
```

Comment:
@detiber, @sdodson, this is what I'm testing.  It's weird but it seems to handle the 1) no docker installed 2) downgrade and 3) upgrade cases.  The idea is that if docker_version_result.stdout is empty we need to install the minimum required version.  If there is a version installed and we didn't need to downgrade then we can ensure the latest version.

I think there's still an edgecase if a version like 1.8.1 was installed.  This is a mess.